packages/adaptive-ui-designer-figma-plugin/src/ui/app.ts

```diff
@@ -627,39 +627,41 @@ export class App extends FASTElement {
     public statesState: StatesState | "unknown" = "unknown";
 
     @observable
-    public selectionDescription: string;
+    public selectionDescription: string = "No selection";
 
     @observable
-    public selectedNodes: PluginUINodeData[] | null;
+    public selectedNodes: PluginUINodeData[] | null = null;
     protected selectedNodesChanged(prev: PluginUINodeData[] | null, next: PluginUINodeData[] | null) {
-        this.controller.selectedNodes = next;
-
-        this.selectionDescription = this.selectedNodes?.map((node) => `${node.type}`).join(" | ") || "No selection";
-
-        this.supportsColor =
-            this.selectedNodes?.some(
-                (node) =>
-                    node.supports.includes(StyleProperty.backgroundFill) ||
-                    node.supports.includes(StyleProperty.foregroundFill) ||
-                    node.supports.includes(StyleProperty.borderFillTop)
-            ) || false;
-        this.supportsBorderThickness = this.controller.supports(StyleProperty.borderThicknessTop);
-        this.supportsDensity = this.controller.supports(StyleProperty.gap);
-        this.supportsCornerRadius = this.controller.supports(StyleProperty.cornerRadiusTopLeft);
-        this.supportsText = this.controller.supports(StyleProperty.fontFamily);
-        this.supportsShadow = this.controller.supports(StyleProperty.shadow);
-
-        this.supportsStyling =
-            this.supportsColor ||
-            this.supportsBorderThickness ||
-            this.supportsDensity ||
-            this.supportsCornerRadius ||
-            this.supportsText ||
-            this.supportsShadow;
-
-        this.supportsDesignSystem = true;
-
-        this.refreshObservables();
+        if (this.controller) {
```

Comment:
Isn't `this.controller` always assigned?

Reply:
It is. I'd have to console log to double check, but it appears that JavaScript process property assignments before the constructor. I don't understand how that could allow the function call, but that Changed handler was firing before `this.controller` was set.

Reply:
The change being that I set the property `null` now instead of leaving it undefined.

Reply:
Oh I see. Yeah order can matter, you would need to define and assign the property prior to defining this property. [This](https://www.typescriptlang.org/play/?#code/MYGwhgzhAEBiD29oG8CwAoaXoAcCuARiAJbDQAWApiCPAFzQQAuATsQHYDm0AvNAORUa8fgG4MGbLkIkyAd3gsQAEwbM2XceknZg8durzAmigBQBKFDqlYm5YhAB0Cpct7QARC5Ufr0AL4Y-kA) is how TypeScript compiles in this scenario